src/context/UsersContext.tsx

```diff
@@ -0,0 +1,52 @@
+import {
+  createContext,
+  Dispatch,
+  FC,
+  ReactNode,
+  SetStateAction,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import { User } from '../types/User';
+import { getUsersData } from '../helpers/getUsersData';
+
+interface Props {
+  users: User[];
+  setActiveUserId: Dispatch<SetStateAction<number | null>>;
```

Comment:
if you wrap your context in memo you should wrap setState function in useCallback, this is not critical, but as a rule, if the state is pushed down, then better use memoization + caching
```suggestion
  setActiveUserId: (userId: number) => void;
```